ConcernsCaseWork/ConcernsCaseWork.Tests/Pages/Case/Management/Action/Nti/AddPageModelTests.cs

```diff
@@ -0,0 +1,114 @@
+﻿using ConcernsCaseWork.Pages.Case.Management.Action.Nti;
+using ConcernsCaseWork.Services.Cases;
+using ConcernsCaseWork.Services.Nti;
+using ConcernsCaseWork.Shared.Tests.Factory;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using Microsoft.AspNetCore.Mvc.Routing;
+using Microsoft.AspNetCore.Mvc.ViewFeatures;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Primitives;
+using Moq;
+using NUnit.Framework;
+using Service.Redis.Nti;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace ConcernsCaseWork.Tests.Pages.Case.Management.Action.Nti
+{
+	[Parallelizable(ParallelScope.All)]
+	public class AddPageModelTests
+	{
+		[Test]
+		public async Task WhenOnGetAsync_MissingCaseUrn_ThrowsException_ReturnPage()
+		{
+			// arrange
+
+			var mockNtiModelService = new Mock<INtiModelService>();
+			var mockNtiReasonsService = new Mock<INtiReasonsCachedService>();
+			var mockNtiStatusesService = new Mock<INtiStatusesCachedService>();
+			var mockLogger = new Mock<ILogger<AddPageModel>>();
+
+			var pageModel = SetupAddPageModel(mockNtiModelService, mockNtiReasonsService, mockNtiStatusesService, mockLogger);
+
+			// act
+			await pageModel.OnGetAsync();
+
+			// assert
+			Assert.That(pageModel.TempData["Error.Message"], Is.EqualTo("An error occurred loading the page, please try again. If the error persists contact the service administrator."));
+		}
+
+		[Test]
+		public async Task WhenOnGetAsync_ReturnsPageModel()
+		{
+			// arrange
+			var mockNtiModelService = new Mock<INtiModelService>();
+			var mockNtiReasonsService = new Mock<INtiReasonsCachedService>();
+			var mockNtiStatusesService = new Mock<INtiStatusesCachedService>();
+			var mockLogger = new Mock<ILogger<AddPageModel>>();
+
+			var pageModel = SetupAddPageModel(mockNtiModelService, mockNtiReasonsService, mockNtiStatusesService, mockLogger);
+
+			var routeData = pageModel.RouteData.Values;
+			routeData.Add("urn", 1);
+
+			// act
+			await pageModel.OnGetAsync();
+
+			// assert
+			mockLogger.Verify(
```

Comment:
This test doesn't actually test what it says it tests - it says it is testing that a PageModel is returned but actually the page could throw an exception and this test would still pass.
It is only testing that a message is logged.